src/components/ProfileUpload/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import useTranslation from 'next-translate/useTranslation'
+import Trans from 'next-translate/Trans'
+
+const ProfileUpload = () => {
+  const [selectedImageUrl, setSelectedImageUrl] = useState<string>('')
+  const router = useRouter()
+  const { t } = useTranslation('sign-up')
+
+  useEffect(() => {
+    // Trigger when refreshing page
+    const handleWindowClose = (event: BeforeUnloadEvent) => {
+      if (!selectedImageUrl) return
+      event.preventDefault()
+
+      return (event.returnValue = t('warningText'))
+    }
+
+    // Trigger when browsing to another page
+    const handleBrowseAway = () => {
+      if (!selectedImageUrl) return
+      const isConfirmed: boolean = window.confirm(t('warningText'))
+
+      if (!isConfirmed) {
+        router.events.emit('routeChangeError')
+        throw 'routeChange aborted.'
+      }
+    }
+
+    window.addEventListener('beforeunload', handleWindowClose)
+    router.events.on('routeChangeStart', handleBrowseAway)
+
+    return () => {
+      window.removeEventListener('beforeunload', handleWindowClose)
+      router.events.off('routeChangeStart', handleBrowseAway)
+    }
+  }, [selectedImageUrl])
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { files } = event.target
+    const hasSingleFile: boolean = files?.length === 1
+
+    if (files && hasSingleFile) {
+      const imageUrl: string = URL.createObjectURL(files[0])
+      setSelectedImageUrl(imageUrl)
+    }
+  }
+
+  const renderProfileImage = () => {
+    const defaultProfileImage: ReactNode = (
+      <img
+        className="profile-upload__image"
+        src="/images/default-profile-image.png"
```

Comment:
Per our last conversation, users have to upload their profile image at the registration because we have to validate their profile before making any transactions.